Sender/BMS_AttributeDataSender.c

```diff
@@ -0,0 +1,91 @@
+  
+/**********************************************
+Include header files
+***********************************************/
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include "BMS_AttributeDataSender.h"
+  
+/**********************************************
+Variable declaration
+***********************************************/
+OperationMode ReadStatus, WriteStatus;
+float Temperature[]={};
+float ChargeRate[]={};
+int BufferArraysize=0;
+
+/*********************************************
+Function definitions
+**********************************************/
+
+/********************************************************************************
+ * Function: BMS_Readfromdatafile
+ 
+ * Description: This function reads the BMS parameter readings from the suitable 
+   data log file and stores into a buffer
+   
+ * input: void
+ 
+ * returns: Read status is Success(True) if the data is successfully read from the file.
+ *********************************************************************************/
+OperationMode BMS_Readfromdatafile()
+{
+  FILE *BMS_datafile;
+  int line=1;
+ ReadStatus= Failure;
+ 
+	
+  BMS_datafile=fopen("./Sender/BMS_attributelog.txt", "r");
+  	if (BMS_datafile==NULL)
+	{
+		printf("File open attempt failed\n");
+		
+	}
+	
+	else
+	{
+		float ReadTemperature=0,ReadChargeRate=0;
+		printf("File open attempt successful\n");
```

Comment:
Your console output would be the input for another program. Did you consider the Receiver complexity, if you are placing English strings in the output?